server/backends/blobstore/aws/aws.go

```diff
@@ -254,3 +256,38 @@ func (a *AwsS3BlobStore) BlobExists(ctx context.Context, blobName string) (bool,
 
 	return true, nil
 }
+
+func (a *AwsS3BlobStore) Writer(ctx context.Context, blobName string) (interfaces.CommittedWriteCloser, error) {
+	// Open a pipe.
+	pr, pw := io.Pipe()
+
+	errch := make(chan error, 1)
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	// Upload from pr in a separate Go routine.
+	go func() {
+		defer pr.Close()
+		_, err := a.uploader.UploadWithContext(
+			ctx,
+			&s3manager.UploadInput{
+				Bucket: a.bucket,
+				Key:    aws.String(util.BlobPath(blobName)),
+				Body:   pr,
+			},
+		)
+		errch <- err
+		close(errch)
+	}()
+
+	committer := &util.CustomCommitter{CommitOp: func() error {
```

Comment:
This functionality is already available via `ioutil.NewCustomCommitWriteCloser`. 
You could have the compressed writer be responsible only for the compression and then wrap it here with additional commit/close funcs. 
```	
    cwc := ioutil.NewCustomCommitWriteCloser(util.NewCompressedBlobStoreWriter(ctx, pw, awsS3Label))
	cwc.CommitFn = func(i int64) error {
		// ...
	}
	cwc.CloseFn = func() error {
		// ...
	}
```